src/uct/ib/rc/base/rc_ep.c

```diff
@@ -45,9 +45,17 @@ UCS_CLASS_INIT_FUNC(uct_rc_ep_t, uct_rc_iface_t *iface)
         goto err_qp_destroy;
     }
 
-    self->unsignaled = 0;
-    self->sl         = iface->config.sl;          /* TODO multi-rail */
-    self->path_bits  = iface->super.path_bits[0]; /* TODO multi-rail */
+    self->unsignaled        = 0;
+    self->sl                = iface->config.sl;          /* TODO multi-rail */
+    self->path_bits         = iface->super.path_bits[0]; /* TODO multi-rail */
+    self->fc_wnd            = iface->config.fc_wnd_size;
+    self->flags             = 0;
+    self->fc_grant_req.func = uct_rc_ep_fc_grant;
```

Comment:
we initialize pending request used for granting credits. If endpoint is not able to grant credits (in response to hard credit request), it adds credit grant to the pending queue.